static/app/components/modals/sentryAppPublishRequestModal.spec.tsx

```diff
@@ -0,0 +1,251 @@
+import type {PropsWithChildren} from 'react';
+import styled from '@emotion/styled';
+import {OrganizationFixture} from 'sentry-fixture/organization';
+import {SentryAppFixture} from 'sentry-fixture/sentryApp';
+
+import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';
+
+import {makeCloseButton} from 'sentry/components/globalModal/components';
+import {
+  getPermissionSelectionsFromScopes,
+  SentryAppPublishRequestModal,
+} from 'sentry/components/modals/sentryAppPublishRequestModal';
+
+describe('SentryAppDetailsModal', function () {
+  const styledWrapper = styled((c: PropsWithChildren) => c.children);
+  const sentryApp = SentryAppFixture();
+
+  afterEach(() => {
+    MockApiClient.clearMockResponses();
+  });
+
+  it('renders the modal', function () {
+    render(
+      <SentryAppPublishRequestModal
+        closeModal={jest.fn()}
+        Header={p => <span>{p.children}</span>}
+        Footer={styledWrapper()}
+        Body={styledWrapper()}
+        CloseButton={makeCloseButton(() => {})}
+        organization={OrganizationFixture()}
+        app={sentryApp}
+      />
+    );
+
+    expect(screen.getByText('Publish Request Questionnaire')).toBeInTheDocument();
+    expect(screen.getByText('Questions to answer')).toBeInTheDocument();
+    expect(screen.getByText('What value does it offer customers?')).toBeInTheDocument();
+    expect(
+      screen.queryByText(
+        'By submitting your integration, you acknowledge and agree that Sentry reserves the right to remove it at any time in its sole discretion.'
+      )
+    ).not.toBeInTheDocument();
+    expect(screen.getByText('Request Publication')).toBeInTheDocument();
```

Comment:
Could you change this to a getByRole to make sure this is accessible? the same way you do on line 124

Reply:
👀 curious Q: wym by accessible ? I guess when should we query by text vs role ?

Reply:
Uhh I'm def not the best person to answer this but basically we want to make sure that components have the correct roles for screen readers, so in this case we want to make sure that "Request Publication" has the role `button`. I thinkkk in most cases it's preferred to query by role if you can?

https://testing-library.com/docs/dom-testing-library/api-accessibility/